src/Console/Util/PhpProcess.php

```diff
@@ -9,85 +9,31 @@
 
 namespace Infection\Console\Util;
 
-use Composer\XdebugHandler\XdebugHandler;
+use Composer\XdebugHandler\PhpConfig;
 use Symfony\Component\Process\Process;
 
 /**
- * Vanilla PHP process and utility functions
- *
  * @internal
  */
 final class PhpProcess extends Process
 {
     /**
-     * @var string|bool
-     */
-    private static $phprc;
-
-    /**
-     * Setups a default Xdebug-free environment for all subprocesses:
+     * Runs a PHP process with xdebug loaded
+     *
+     * If xdebug was loaded in the main process, it will have been restarted
+     * without xdebug and configured to keep xdebug out of PHP sub-processes.
      *
-     * - PHPRC should point to our temporary php.ini (we need to save a previous value)
+     * This method allows a sub-process to run with xdebug enabled (if it was
+     * originally loaded), then restores the xdebug-free environment.
      *
-     * - PHP_INI_SCAN_DIR should be made blank because our php.ini has all we need
-     *   (a previous value can be found in XdebugHandler::getRestartSettings()["scanDir"])
+     * {@inheritdoc}
      */
-    public static function setupXdebugFreeEnvironment(): void
-    {
-        if (!isset(self::$phprc)) {
-            self::$phprc = getenv('PHPRC');
-        }
-
-        self::putenv('PHPRC', XdebugHandler::getRestartSettings()['tmpIni']);
-        self::putenv('PHP_INI_SCAN_DIR', '');
-    }
-
     public function start(callable $callback = null, array $env = null): void
     {
-        // Xdebug wasn't skipped, running as is
-        if ('' === XdebugHandler::getSkippedVersion()) {
-            parent::start($callback, $env ?? []);
-
-            return;
-        }
-
-        /*
-         * Vanilla processes are expected to run in a vanilla, or the most original, environment.
-         *
-         * For that, we need to remove or reset all environment variables we set to setup our custom
-         * xdebug-free environment, later setting them back so we won't have to think again about it
-         * for the bulk of other processes, which not require xdebug to function, and work better
-         * without it.
-         */
```

Comment:
I feel like this comment is still relevant, for one reason because it isn't immediately clear what is going on here. Only it could go somewhere else, like before `new PhpConfig` or around that.

Reply:
I tried to cover that remark, which I agree is needed, in the actual method doc-comment.

Reply:
I actually wondered if it was worth renaming the class to something like `PhpXdebugProcces` to make things clearer?

Reply:
`PhpXdebugProcces` seems to be better than just `PhpProcces`, but if we factor in that we could be running under phpdbg, it could become slightly confusing (no Xdebug will be added in that case).

Almost everything is fine with that doc-comment, but it lost some of the big picture explanation (as to why we do that). 

> This method allows a sub-process to run with xdebug enabled (if it was
> originally loaded in the vanilla environment), then restores the xdebug-free 
> environment. Therefore we won't have to think again about it for the bulk 
> of other processes, which not require nor need xdebug to function, and 
> work better without it.

How would you think if we amend the last paragraph like so?

Reply:
Sure, I'll add that.